bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigTomlParser.java

```diff
@@ -59,34 +56,34 @@
  */
 public class ConfigTomlParser {
 
-    static final Path CONFIG_FILE_PATH = Paths.get(RuntimeUtils.USER_DIR).resolve(CONFIG_FILE_NAME);
-
     private ConfigTomlParser() {
     }
 
-    private static TomlTableNode getConfigurationData() throws TomlException {
-        if (!Files.exists(CONFIG_FILE_PATH)) {
-            throw new TomlException("Configuration toml file `" + CONFIG_FILE_NAME + "` is not found");
+    private static TomlTableNode getConfigurationData(Path filePath) throws TomlException {
+        Path configFilePath = filePath.resolve(CONFIG_FILE_NAME);
+        if (!Files.exists(configFilePath)) {
+            throw new TomlException("Configuration toml file `" + CONFIG_FILE_NAME + "` is not found in directory '" +
+                    filePath.toString() + "'");
         }
-        ConfigToml configToml = new ConfigToml(CONFIG_FILE_PATH);
+        ConfigToml configToml = new ConfigToml(configFilePath);
         return configToml.tomlAstNode();
     }
 
-    public static void populateConfigMap(Map<Module, VariableKey[]> configurationData) throws TomlException {
+    public static void populateConfigMap(Path filePath, Map<Module, VariableKey[]> configurationData)
+            throws TomlException {
         if (configurationData.isEmpty()) {
             return;
         }
-        TomlTableNode tomlNode = getConfigurationData();
+        TomlTableNode tomlNode = getConfigurationData(filePath);
         if (tomlNode.entries().isEmpty()) {
             //No values provided at toml file
             return;
         }
         for (Map.Entry<Module, VariableKey[]> moduleEntry : configurationData.entrySet()) {
-            String orgName = moduleEntry.getKey().getOrg();
-            String moduleName = moduleEntry.getKey().getName();
-            TomlTableNode orgNode = orgName.equals(ANON_ORG) ? tomlNode : extractOrganizationNode(tomlNode, orgName);
-            TomlTableNode moduleNode = moduleName.equals(DEFAULT_MODULE) ? orgNode : extractModuleNode(orgNode,
-                    moduleName);
+            TomlTableNode moduleNode = retrieveModuleNode(tomlNode, moduleEntry.getKey());
+            if (moduleNode == null) {
```

Comment:
Why can this be null? retrieveModuleNode method inside throw a exception if it null.